v2.2/architecture/life-of-a-distributed-transaction.md

```diff
@@ -0,0 +1,204 @@
+
+These dismantled `BatchRequests` are reassembled into new `BatchRequests` containing the address of the range's leaseholder.
+
+All writes operations also propagate the leaseholder's address back to the `TxnCoordSender`, so it can track and clean up write operations as necessary.
+
+The `DistSender` sends out the first `BatchRequest` for each range in parallel. As soon as it receives a provisional acknowledgment from the leasholder node’s evaluator (details below), it sends out the next `BatchRequest` for that range.
+
+The DistSender then waits to receive commit acknowledgments for all of its write operations, as well as values for all of its read operations.
+
+<!-- #### Finding leaseholders (meta ranges)
+
+Each range is replicated at least three times, and one of these range is chosen as the leaseholder, which serves and reads and coordinates all writes for the range. Because of this architecture, all of the KV operations the `DistSender` receives must be routed directly to the range's leaseholder.
+
+To determine which physical node to send the operation to, the `DistSender` relies on data from the cluster's meta ranges, which detail which physical nodes contain the range's replicas as well as which node is the leaseholder. For any given request, the `DistSender` can determine the leaseholders by performing a read from the meta ranges for the keys in question.
+
+For example, if the `DistSender` needs to route a request for keys `[1000, 2000)`, it performs a read across the meta ranges for those values. The response contains the leaseholder as the first node in the response.
+
+To optimize performance, nodes heavily cache the values they've previously seen in meta ranges, which means that in most cases this read is served by a local cache, rather than going all the way out to the network. -->
+
+<!-- #### Transaction Record
+
+The first range that a transaction touches receives an additional KV operation to create a translation record (`TxnRecord`). This key-value pair is used to store the transaction's current state for the life of the transaction. It starts off in a `PENDING` state, and then progresses to either `COMMITTED` if the transaction succeeds or `ABORTED` if the transaction must be canceled.  -->
+
+## Leaseholder node
+
+The gateway node's `DistSender` tries to send its `BatchRequests` to the replica identified as the range's [leaseholder](replication-layer.html#leases), which is a single replica that serves all reads for a range, as well as coordinates all writes. Leaseholders play a crucial role in CockroachDB's architecture, so it's a good topic to make sure you're familiar with.
+
+### Request response
+
+Because the leaseholder replica can shift between nodes, all nodes must be able to return a request for any key.
+
+##### No Longer Leaseholder
+
+If a node:
+
+- *Was* the leaseholder, but no longer holds the lease
+- Holds a replica for the range
```

Comment:
Does not hold a replica for the range? I'm not sure what you're trying to enumerate here.